src/scheduler.cc

```diff
@@ -223,7 +224,7 @@ Status SchedulerService::TaskInfo(ServerContext* context, const TaskInfoRequest*
   }
   for (int i = 0; i < workers->size(); ++i) {
     OperationId operationid = (*workers)[i].current_task;
-    if (operationid != NO_OPERATION) {
+    if (operationid != NO_OPERATION && operationid != ROOT_OPERATION) {
```

Comment:
Actually not 100% sure about this, it might not remove the need for manual checking in slightly different situations. Thoughts?